test/cluster/util.py

```diff
@@ -657,3 +658,46 @@ def get_replica_count(rf: ReplicationOption) -> int:
         get_replica_count(["2"]) == 2
     """
     return len(rf) if type(rf) is list else int(rf)
+
+
+async def wait_for_keyspace_rf_change_to_complete(cql, ks_name: str, timeout: float = 120.0, period: float = 1.0) -> None:
+    """
+    Waits for any ongoing replication factor change for the specified keyspace to complete.
+    """
+    deadline = time.time() + timeout
+
+    async def no_pending_rf_change():
+        rows = await cql.run_async(
+            "SELECT request_type, done, new_keyspace_rf_change_ks_name FROM system.topology_requests " \
+            f"WHERE request_type = 'keyspace_rf_change' AND  new_keyspace_rf_change_ks_name = '{ks_name}' ALLOW FILTERING"
+        )
+        for row in rows:
+            if not row.done:
+                return None
+        return True
+
+    await wait_for(no_pending_rf_change, deadline=deadline, period=period)
+
+
+async def run_statement_with_rf_retry(cql, ks_name: str, statement: str, attempts: int = 3, simulate_error: bool = False) -> None:
+    """
+    Runs the given CQL statement, retrying if there is an ongoing replication factor change
+    """
+    while attempts > 0:
+        try:
+            if simulate_error:
+                simulate_error = False
+                raise InvalidRequest("Another RF change for this keyspace is already in progress (test_case)")
```

Comment:
I do not understand this functionality. It is a test side function, we are not simulating error for anyone real using the DB, only for our test. If we know we want this error to trigger, we can trigger it in the test itself.

Not to mention the fact that it conceptually doesnt make sense to include in `run_statement_with_rf_retry`, which can be run with Any statement and this only returns errors for RF change, not any other error.

Reply:
yes it was just my internal testing i will remove , but im not sure that this change will even be pushed